### PR TITLE
chore: add ssr migration for 3.0

### LINKED ISSUE #10183: Schematics migration should add the Setup library
The new for 3.0 `@spartacus/setup` library contains recipes and ssr. Since the SSR features from the Setup libraries were available in `@spartacus/core` before, schematics should detect this usage and automatically at the Setup library as import.

The current schematics version does this:
```
// TODO:Spartacus - 'NgExpressEngineDecorator' was moved to @spartacus/setup/ssr.
import { NgExpressEngineDecorator } from '@spartacus/core';
``` 

- It should add the new library and change the import (if possible)

### PR DESCRIPTION
**Pull request by znikola**
closes #10183

### CODE REVIEW

File: projects/schematics/src/migrations/2_0/ssr/ssr.ts [hunk 1]
```diff
@@ -298,28 +299,6 @@ export function updateAngularJsonFile(): Rule {
   };
 }
 
-async function checkIfSSRIsUsed(tree: Tree): Promise<boolean> {
```
Comment (LTiger14) [inline projects/schematics/src/migrations/2_0/ssr/ssr.ts:301]: Why did you remove it for 2.0?

File: projects/schematics/src/shared/utils/package-utils.ts [hunk 2]
```diff
@@ -51,3 +53,25 @@ export function isAngularLocalizeInstalled(tree: Tree): boolean {
...
+  const buffer = tree.read('angular.json');
+  if (!buffer) {
+    throw new SchematicsException('Could not find angular.json');
+  }
+  const angularFileBuffer = buffer.toString(UTF_8);
+  const angularJson = JSON.parse(angularFileBuffer);
+  const isServerConfiguration = !!angularJson.projects[projectName].architect[
+    'server'
+  ];
+  const serverFilePath = getPathResultsForFile(tree, 'server.ts', '/')[0];
```
Comment (LTiger14) [inline projects/schematics/src/shared/utils/package-utils.ts]: Apparently some customers have their server in a file not named `server.ts` or in a different location. For example, it could be `server/index.ts`.

This file can be found in the `angular.json` "server configuration" in "options" -> "main". Maybe we could use that value instead of a hard-coded file path.

What do you think?

File: projects/schematics/src/migrations/3_0/ssr/ssr.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          addSetupPackageJsonDependencies(packageJson),
+          installPackageJsonDependencies(),
+        ])
+      : noop();
+  };
+}
+
+function updateImport(): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const serverFilePath = getPathResultsForFile(tree, 'server.ts', '/')[0];
```
Comment (LTiger14) [inline projects/schematics/src/migrations/3_0/ssr/ssr.ts]: See comment bellow ⬇️

File: projects/schematics/src/migrations/2_0/ssr/ssr.ts (hunk 1 above)
Comment (znikola) [reply projects/schematics/src/migrations/2_0/ssr/ssr.ts:301]: Not removed, moved to a common location in the `projects/schematics/src/shared/utils/package-utils.ts`

File: projects/schematics/src/shared/utils/package-utils.ts (hunk 2 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/package-utils.ts]: Good point, let's do that.

File: projects/schematics/src/migrations/2_0/ssr/ssr.ts (hunk 1 above)
Comment (LTiger14) [reply projects/schematics/src/migrations/2_0/ssr/ssr.ts:301]: Good I thought it looked similar 😄

**Review by LTiger14 — APPROVED** [review]
